seungjong/app.js

```diff
@@ -47,11 +47,74 @@ app.post('/signUp', async (req, res, next) => {
             profile_image
         ) VALUES (?, ?, ?, ?);
         `,
-        [name, email, password, profileImage]
+        [name, email, password, prfileImage]
     );
-    res.status(201).json({ message: "sucessfully created!" });
+    res.status(200).json({ message: "sucessfully created!" });
 })
 
+app.post('/addPost', async (req, res, next) => {
+    const { title, content, userId } = req.body;
+
+    await dataSource.query(
+        `INSERT INTO posts (
+            title,
+            content,
+            user_id)
+            VALUES (?, ?, ?);
+            `, [title, content, userId]
+    );
+    res.status(201).json({ message: "postCreated!" });
+});
+
+app.get('/posts', async (req, res, next) => {
+    const { userId, userProfileImage, postingId, postinImageUrl, postingContent } = req.body;
+
+    await dataSource.query(`
+        SELECT
+        u.id,
+        u.profile_image,
+        p.user_id,
+        p.image_url,
+        p.content
+        FROM
+        users as u
+        INNER JOIN posts as p
+        ON u.id = p.user_id
+        ;
+        `
+        , (err, rows) => {
+            res.status(200).json(rows);
+        })
+})
+
+app.get('/users/:userId/posts', async (req, res, next) => {
+    const { userId } = req.params;
+    const posts = await dataSource.query(
+        `SELECT
+            users.id as userId,
+            users.profile_image as userProfileImage,
+            (
+                SELECT
+                JSON_ARRAYAGG(
+                    JSON_OBJECT(
+                        "postingId", posts.id,
+                        "postingImageUrl", posts.image_url,
+                        "postingContent", posts.content
+                    )
+                )
+                FROM posts
+                JOIN users ON users.id = posts.user_id
+                WHERE posts.user_id = ?
+            ) as postings
+            FROM users
+            WHERE users.id = ?;`, [userId, userId]
+    )
+    res.status(200).json({message: "성공", userId, data: posts});
```

Comment:
```suggestion
    res.status(200).json({message: "성공", data: posts});
```
posts 변수 즉, SELECT query에서 받아오는 데이터중에, userId가 이미 있기 때문에 해당 코드는 수정되어야할 것 같습니다.